### DISCUSSION TITLE
Q: Return dynamic API response in a browser

### DISCUSSION THREAD

**Discussion by sagarsoni7** [requestly/requestly#1152] [Q&A]
Question: 

Can you give an example that how can I use Dynamic response by writing javascript method? 
This is my need => i have two services , they are X and Y. 
I want if Y services call and gives 200, make X reponse an area false. If not ,make ture. 
Can you give an example?

Asked by: identity_hidden

**Comment by nsrCodes** [comment]
It is currently not possible to directly share rule execution metadata between separate rules. But there are some ways of achieving this depending on the type of service that you have and the nature of your app:

# Approach 1: Make the request to service Y yourself
#### Assumption: Service Y is assumed to always be called, and Y is idempotent (meaning results from the API do not change when the same request is made again)

in that case you can apply a rule on the call to service X, inside which you can make the request to Y and modify the response from X accordingly. Here's [a sample rule](https://app.requestly.io/rules#sharedList/1697021292698-make-Request-to-Y-while-modifying-request-to-X)
<img width="880" alt="Screenshot 2023-10-11 at 4 19 06 PM" src="https://github.com/requestly/requestly/assets/57226514/fcbf7706-d3e5-44fa-8c60-1a04a14b3c4f">

# Approach 2: share state between rule execution by saving state in window object. 
#### Assumption 1: Call for service Y is made before any calls to service X
##### Note: This works only on the extension, not the desktop app

Currently when programmatically modifying the response, the status code is not being passed inside the arguments of modification code. This should be easy to add and [I have raised an issue for this](https://github.com/requestly/requestly/issues/1154). Meanwhile if you have any extra parameters indicating the response was succesfull (for eg, by analyzing the response body), you  can set a variable inside the window object, like `wasYSuccessfull` to true,

Then inside the rule for X check if `window.wasYSuccessfull` is true, and modify the body accordingly.

Here are [sample rules](https://app.requestly.io/rules#sharedList/1697024165130-Co-Dependent-rules-sharing-state-with-window-object) for these:

<img width="565" alt="Screenshot 2023-10-11 at 5 07 01 PM" src="https://github.com/requestly/requestly/assets/57226514/ddebe0e1-7fef-4fc7-bbe2-af62eac0a3b2">
<img width="565" alt="Screenshot 2023-10-11 at 5 09 43 PM" src="https://github.com/requestly/requestly/assets/57226514/f004f8ef-0b47-48ab-a968-ad81ae065530">
